src/OrleansTelemetryConsumers.DataDog/DataDogTelemetryConsumer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Newtonsoft.Json;
+using Orleans.Runtime;
+using StatsdClient;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orleans.TelemetryConsumers.DataDog
+{
+    public class DataDogTelemetryConsumer : IMetricTelemetryConsumer, IEventTelemetryConsumer
+    {
+        public DataDogTelemetryConsumer(string statsdServerName = "127.0.0.1", int statsdPort = 8125, string prefix = "")
+        {
+            var dogstatsdConfig = new StatsdConfig
+            {
+                StatsdServerName = statsdServerName,
+                StatsdPort = statsdPort,
+                Prefix = prefix
+            };
+
+            DogStatsd.Configure(dogstatsdConfig);
+        }
+
+        public void DecrementMetric(string name)
+        {
+            DogStatsd.Decrement(name);
+        }
+
+        public void DecrementMetric(string name, double value)
+        {
+            DogStatsd.Decrement(name, (int)value);
+        }
+
+        public void IncrementMetric(string name)
+        {
+            DogStatsd.Increment(name);
+        }
+
+        public void IncrementMetric(string name, double value)
+        {
+            DogStatsd.Increment(name, (int)value);
+        }
+
+        public void TrackMetric(string name, TimeSpan value, IDictionary<string, string> properties = null)
+        {
+            DogStatsd.Increment(name, (int)value.TotalMilliseconds, tags: ToStringArray(properties));
+        }
+
+        public void TrackMetric(string name, double value, IDictionary<string, string> properties = null)
+        {
+            DogStatsd.Increment(name, (int)value, tags: ToStringArray(properties));
```

Comment:
I believe here are a few potential issues here:
1. The range of double is larger than that of int. With a quick search on what would happen, I found [this](http://stackoverflow.com/questions/4181942/convert-double-to-int).
2. Cutting of the fractional parts introduce bias as described at [this SO post](http://stackoverflow.com/questions/311696/why-does-net-use-bankers-rounding-as-default). How about [round-to-even](https://msdn.microsoft.com/en-us/library/system.math.round.aspx) and then to either convert or cast, albeit there is the issue as in 1.

Why is this cast needed, by the way?